dataset_builder/humaneval_to_kt.py

```diff
@@ -0,0 +1,179 @@
+                    return translate_type(slice) + "?"
+                case other:
+                    raise Exception(f"Bad generic {other}")
+        case ast.Name("int") | "int":
+            return "Int"
+        case ast.Name("float"):
+            return "Double"
+        case ast.Name("bool"):
+            return "Boolean"
+        case ast.Name("str") | "str":
+            return "String"
+        case None:
+            raise Exception("No None class in Kotlin")
+        # nothing was changed below this comment
+        case ast.Name("Any"):
+            return "any"
+        case ast.Name(x):
+            raise Exception(f"unknown name {x}")
+        case ast.Constant(Ellipsis):
+            raise Exception("No ellipses!")
+        case _other:
+            raise Exception(f"unknown annotation: {t}")
+
+TargetExp = str
+
+def coerce(expr: str, type) -> str: 
+    match expr, type:
+        case _: 
+            return expr
+
+class Translator:
+
+    stop = [ '\nfun ', '\n/*', '\n//', '\nclass' ]
+
+    def __init__(self):
+        global needs_hashmap
+        self.type = None
+        self.is_candidate_result = False
+
+    def file_ext(self):
+        return "kt"
+
+    # NOTE: this definition translates the prompt but not the examples in them, the examples provided are very much still in Python syntax
+    def translate_prompt(self, name: str, args: List[ast.arg], returns, description: str) -> str:
+        global needs_hashmap
+        # added this line to substitute candidate in for entry point later on
+        self.entry_point = to_camel_case(name)
+        js_description = "// " + re.sub(DOCSTRING_LINESTART_RE, "\n// ", description.strip().replace(name, self.entry_point)) + "\n"
+        # Store this for later coercions on tests
+        needs_hashmap = False
+        self.type = [[arg.annotation for arg in args], returns]
+        def translate_arg(arg):
+            return arg.arg + ": " + translate_type(arg.annotation)
+        arg_strings = []
+        return_type = ""
+        try:
+            arg_strings = [translate_arg(arg) for arg in args]
+            return_type = translate_type(returns)
+        except Exception as e:
+            print(e)
+            return None
+        arg_list = ", ".join(arg_strings)
+        return f"{js_description}fun {to_camel_case(name)}({arg_list}): {return_type} "+"{\n"
+
+    def test_suite_prefix_lines(self, entry_point) -> List[str]:
+        """
+        This code goes at the start of the test suite.
+        """
+        return ["fun main() {"]
+
+    def test_suite_suffix_lines(self) -> List[str]:
+        return ["}"]
+
+    def deep_equality(self, left: str, right: str) -> str:
+        """
+        All tests are assertions that compare deep equality between left and right.
+
+        Make sure you use the right equality operator for your language. For example,
+        == is the wrong operator for Java and OCaml.
+        """
+        if self.is_candidate_result:
```

Comment:
Why is this flag needed?

Reply:
Honestly, am not sure, I edited this translator from the Typescript translator and avoided touching the parts I did not fully understand.

Reply:
It looks like the Julia translator (`humaneval_to_jl.py`) is using something like this to correctly translate an empty list literal in the testing function. I think you should be able to get rid of it unless you need to have something similar.